### ISSUE TITLE
Difficult to construct client with no config

### ISSUE THREAD

**Issue by Chouser** [mcohen01/amazonica#267]
SimpleWorkflowClient uses long polling and requires client timeouts of greater than 60 seconds to behave correctly. A SimpleWorkflowClient constructed with no parameters uses a special [default config](https://github.com/aws/aws-sdk-java/blob/7844c64cf248aed889811bf2e871ad6b276a89ca/aws-java-sdk-simpleworkflow/src/main/java/com/amazonaws/services/simpleworkflow/AmazonSimpleWorkflowClientConfigurationFactory.java#L30) that does the Right Thing. However, Amazonica apparently always builds and provides a config, and this apparently overrides the one SimpleWorkflowClient needs.

Our workaround was to manually re-apply the SWF defaults:
```
(amz-swf/poll-for-decision-task
  (assoc amz-creds                                
    :client-config {:socket-timeout 90000    
                    :max-connections 1000}))
  params)
```

If passing no-config isn't desirable, another option would be to allow passing the already-built config object provided by AmazonSimpleWorkflowClientConfigurationFactory, but that doesn't seem supported either.

Finally, while tracking this down we noticed the [consturctors for AmazonSimpleWorkflowClient are now all deprecated](https://github.com/aws/aws-sdk-java/blob/a90a49d088b5afadba048efe52f7d57652cb11ed/aws-java-sdk-simpleworkflow/src/main/java/com/amazonaws/services/simpleworkflow/AmazonSimpleWorkflowClient.java#L128) in favor of AmazonSimpleWorkflowClientBuilder

Thanks for all your great work on Amazonica!

**Comment by mcohen01** [comment]
I believe in the latest release you can do something like:

```clojure
(import com.amazonaws.services.simpleworkflow.AmazonSimpleWorkflowAsyncClientBuilder)
(def creds {:client-config
  {:amazon-client-fn
    (fn [_ _ _] (AmazonSimpleWorkflowAsyncClientBuilder/defaultClient))}})
(amz-swf/poll-for-decision-task creds)
```

**Comment by mcohen01** [comment]
this should be fixed with `0.3.100`
